internal/server/repository_workerauth.go

```diff
@@ -879,22 +919,55 @@ func decrypt(ctx context.Context, value []byte, wrapper wrapping.Wrapper) ([]byt
 	return marshaledInfo, nil
 }
 
-// FindWorkerAuthByWorkerId takes a workerId and returns the WorkerAuth record associated with that worker.
-func (r *WorkerAuthRepositoryStorage) FindWorkerAuthByWorkerId(ctx context.Context, workerId string) (*WorkerAuth, error) {
+// FindWorkerAuthByWorkerId takes a workerId and returns the WorkerAuthSet for this worker.
+func (r *WorkerAuthRepositoryStorage) FindWorkerAuthByWorkerId(ctx context.Context, workerId string) (*WorkerAuthSet, error) {
 	const op = "server.(WorkerAuthRepositoryStorage).FindWorkerAuthByWorkerId"
 	if len(workerId) == 0 {
 		return nil, errors.New(ctx, errors.InvalidParameter, op, "empty worker ID")
 	}
 
-	worker := allocWorkerAuth()
-	worker.WorkerId = workerId
-	err := r.reader.SearchWhere(ctx, &worker, "worker_id = ?", []interface{}{workerId})
-	if err != nil {
-		if errors.Is(err, dbw.ErrRecordNotFound) {
-			return nil, nil
-		}
+	var previousWorkerAuth *WorkerAuth
+	var currentWorkerAuth *WorkerAuth
+
+	var workerAuths []*WorkerAuth
+	if err := r.reader.SearchWhere(ctx, &workerAuths, "worker_id = ?", []interface{}{workerId}); err != nil {
 		return nil, errors.Wrap(ctx, err, op)
 	}
 
-	return worker, nil
+	workerAuthsFound := len(workerAuths)
+	if workerAuthsFound == 0 {
```

Comment:
This would be cleaner as a `switch` function on `workerAuthsFound`, with the `default` being the >2 case.

Reply:
thanks!